src/utils.ts

```diff
@@ -67,27 +74,34 @@ export function isTrueJoinEvent(event: any): boolean {
     return membership === "join" && prevMembership !== "join";
 }
 
-/**
- * Redact a user's messages in a set of rooms.
- * See `getMessagesByUserIn`.
- *
- * @param client Client to redact the messages with.
- * @param managementRoom Management room to log messages back to.
- * @param userIdOrGlob A mxid or a glob which is applied to the whole sender field of events in the room, which will be redacted if they match.
- * See `MatrixGlob` in matrix-bot-sdk.
- * @param targetRoomIds Rooms to redact the messages from.
- * @param limit The number of messages to redact from most recent first. If the limit is reached then no further messages will be redacted.
- * @param noop Whether to operate in noop mode.
- */
-export async function redactUserMessagesIn(
+async function adminRedactUserMessagesIn(
+    client: MatrixSendClient,
+    managementRoom: ManagementRoomOutput,
+    userId: string,
+    targetRooms: string[],
+    limit = 1000,
+) {
+    let redactID: string;
+    const body = { limit: limit, rooms: targetRooms };
+    const redactEndpoint = `/_synapse/admin/v1/user/${userId}/redact`;
+    const response = await client.doRequest("GET", redactEndpoint, null, body);
+    redactID = response["redact_id"];
```

Comment:
```suggestion
    const body = { limit: limit, rooms: targetRooms };
    const redactEndpoint = `/_synapse/admin/v1/user/${userId}/redact`;
    const response = await client.doRequest("GET", redactEndpoint, null, body);
    const redactID = response["redact_id"];
```

if it's complaining about types, use `response["redact_id"] as string`